Original patch by Luke-Jr (luke-jr+gentoobugs@utopios.org)
Minor fixes by Manuel Rueger (mrueg@rueg.eu)

https://bugs.gentoo.org/show_bug.cgi?id=216114
Gentoo specific additions

--- Makefile
+++ Makefile
@@ -23,7 +23,6 @@
 	$(INSTALL_DATA) $(CONFFILE) $(DESTDIR)$(etcdir)/etckeeper/etckeeper.conf
 	$(INSTALL_EXE) etckeeper $(DESTDIR)$(bindir)/etckeeper
 	$(INSTALL_DATA) etckeeper.8 $(DESTDIR)$(mandir)/man8/etckeeper.8
-	$(INSTALL_DATA) bash_completion $(DESTDIR)$(etcdir)/bash_completion.d/etckeeper
 ifeq ($(HIGHLEVEL_PACKAGE_MANAGER),apt)
 	$(INSTALL_DATA) apt.conf $(DESTDIR)$(etcdir)/apt/apt.conf.d/05etckeeper
 	mkdir -p $(DESTDIR)$(etcdir)/cruft/filters-unex
@@ -39,7 +37,6 @@
 ifeq ($(HIGHLEVEL_PACKAGE_MANAGER),zypper)
 	$(INSTALL_DATA) zypper-etckeeper.py $(DESTDIR)$(prefix)/lib/zypp/plugins/commit/zypper-etckeeper.py
 endif
-	-./etckeeper-bzr/__init__.py install --root=$(DESTDIR) ${PYTHON_INSTALL_OPTS} || echo "** bzr support not installed"
 	echo "** installation successful"
 
 clean: etckeeper.spec
--- etckeeper.conf
+++ etckeeper.conf
@@ -31,11 +31,13 @@
 
 # The high-level package manager that's being used.
 # (apt, pacman-g2, yum, zypper etc)
-HIGHLEVEL_PACKAGE_MANAGER=apt
+# For gentoo this is emerge
+HIGHLEVEL_PACKAGE_MANAGER=emerge
 
 # The low-level package manager that's being used.
 # (dpkg, rpm, pacman, pacman-g2, etc)
-LOWLEVEL_PACKAGE_MANAGER=dpkg
+# For gentoo this is qlist
+LOWLEVEL_PACKAGE_MANAGER=qlist
 
 # To push each commit to a remote, put the name of the remote here.
 # (eg, "origin" for git).
--- list-installed.d/50list-installed
+++ list-installed.d/50list-installed
@@ -9,4 +9,6 @@
 	rpm -qa --qf "%|epoch?{%{epoch}}:{0}|:%{name}-%{version}-%{release}.%{arch}\n" | sort
 elif [ "$LOWLEVEL_PACKAGE_MANAGER" = pacman ]; then
 	pacman -Q
+elif [ "$LOWLEVEL_PACKAGE_MANAGER" = qlist ]; then
+	qlist -ICv
 fi
--- update-ignore.d/01update-ignore
+++ update-ignore.d/01update-ignore
@@ -99,6 +99,10 @@
 		ignore "*.pacorig"
 		ignore "*.pacsave"
 		nl
+	elif [ "$LOWLEVEL_PACKAGE_MANAGER" = "qlist" ]; then
+		comment "new and old versions of conffiles, stored by emerge"
+		ignore "*._cfg*"
+		nl
 	fi
 	
 	comment "old versions of files"
